### PR TITLE
test: add test case for nacos service de-duplicate

### PR DESCRIPTION
**Pull request by tzssangglass**
the test cases of https://github.com/apache/apisix/pull/6247

Signed-off-by: tzssangglass <tzssangglass@gmail.com>

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by tzssangglass** [comment]
Since #6247 has not been updated for a long time, I decided to fix it in this PR.

File: t/discovery/nacos2.t [hunk 1]
```diff
@@ -112,3 +112,105 @@ GET /hello
...
+
+            local body = json_decode(res.body)
+            local services = body.services
+            local service = services["public"]["DEFAULT_GROUP"]["APISIX-NACOS"]
+            local number = table.getn(service)
+            ngx.say(number)
+        }
+    }
+--- response_body
+2
```
Comment (spacewander) [inline t/discovery/nacos2.t:216]: Should be 1 after deduplicating?

Comment (tzssangglass) [reply t/discovery/nacos2.t:216]: In fact, it is 2 instances of the same service. In docker-compose, the 2 service instances are started with the same namespace, group and service name, but have different exposed ports, which is equivalent to different instances of the same service registered to nacos.

docker-compose:
https://github.com/apache/apisix/blob/48398d3f1501fe425b08673609605ea9d8039f54/ci/pod/docker-compose.yml#L230-L263

these 2 service instances on nacos-dashboard:

51646108152_ pic

Before de-duplication, the number of APISIX fetches to these two services is 6:
https://github.com/apache/apisix/runs/5180101415?check_suite_focus=true
61646108418_ pic

6 = 2 instances * ( in route + in service + in upstream )

**Review by spacewander — APPROVED** [review]
